glocaltokens/client.py

```diff
@@ -405,6 +411,15 @@ def find_device(unique_id: str) -> NetworkDevice | None:
                         unique_id,
                     )
                     network_device = find_device(unique_id)
+                elif item.device_name in discovery_address_dict:
+                    network_device = NetworkDevice(
+                        name=item.device_name,
+                        ip_address=discovery_address_dict[item.device_name]
+                        or "0.0.0.0",
```

Comment:
I still believe we should not default the IP to 0.0.0.0.
According to wikipedia

> the address 0.0. 0.0 is a non-routable meta-address used to designate an invalid, unknown or non-applicable target

We are getting `discovery_address_dict` from the user so it must contain name and IP of devices, right? If you want to make it extra rigid we can add an inout validation to the `get_google_devices` function to check the data in `discovery_address_dict` to contain expected name/ip pairs, so if the input `dict` is invalid we raise the error :)

```suggestion
                        ip_address=discovery_address_dict[item.device_name],
```

Reply:
I think it is a great idea to validate the input

Reply:
I have added input validation to the dict, it also checks if the IP address is a valid IPv4 address 😄